core/kazoo_attachments/src/gen_attachment.erl

```diff
@@ -0,0 +1,84 @@
+-module(gen_attachment).
+
+%% Setter
+-export([error_response/2]).
+%% Getters
+-export([error_code/1, error_body/1]).
+%% Helper(s)
+-export([is_error_response/1]).
+
+-include("kz_att.hrl").
+
+-type settings()         :: kz_data:connection().
+-type db_name()          :: kz_term:ne_binary().
+-type doc_id()           :: kz_term:ne_binary().
+-type att_name()         :: kz_term:ne_binary().
+-type contents()         :: kz_term:ne_binary().
+-type options()          :: kz_data:options().
+-type handler_props()    :: kz_data:connection().
+-type error_code()       :: pos_integer() | atom(). % 400, 404, 409, etc.
+-type error_body()       :: binary() | % encoded map()
+                            bitstring() | % <<"example">>
+                            atom(). % 'not_found' | 'return_id_missing' | etc.
+-opaque error_response() :: {'error', [{'error_code', error_code()} |
```

Comment:
Why not have an error tuple instead of a list?

```erlang
-type error_response() :: {'error', error_code(), error_body()}.
```

Since you don't want modules matching on the error (which seems wrong but whatever).